cmd/iam.go

```diff
@@ -215,27 +216,38 @@ func (p *awsS3Provisioner) createBucketPolicyDocument(bktName string) (string, e
 		Statement: []StatementEntry{},
 	}
 
-	// do a switch case here to figure out which policy to include
-	// for now we are commenting until we can update the lib
-	// this will come from bucketOptions I'm guessing (obc or sc params)?
-	/*
-		if spec.LocalPermission != nil {
-			switch *spec.LocalPermission {
-			case storageV1.ReadOnlyPermission:
-				policy.Statement = append(policy.Statement, read)
-			case storageV1.WriteOnlyPermission:
-				policy.Statement = append(policy.Statement, write)
-			case storageV1.ReadWritePermission:
-				policy.Statement = append(policy.Statement, read, write)
-			default:
-				return "", fmt.Errorf("unknown permission, %s", *spec.LocalPermission)
-			}
+	// Check if the storage class has provided a storage policy we should use...
+	if p, ok := options.Parameters["iamPolicy"]; ok {
+		err := json.Unmarshal([]byte(p), &policy)
+		if err != nil {
+			return "", err
+		}
+		// Ensure each policy is tied to just this bucket
+		for idx, _ := range policy.Statement {
+			policy.Statement[idx].Resource = []string{arn + "/*", arn}
 		}
-	*/
-	//For now hard coding read and write
-	policy.Statement = append(policy.Statement, read, write)
+	} else {
+		// do a switch case here to figure out which policy to include
+		// for now we are commenting until we can update the lib
+		// this will come from bucketOptions I'm guessing (obc or sc params)?
+		/*
+			if spec.LocalPermission != nil {
+				switch *spec.LocalPermission {
+				case storageV1.ReadOnlyPermission:
+					policy.Statement = append(policy.Statement, read)
+				case storageV1.WriteOnlyPermission:
+					policy.Statement = append(policy.Statement, write)
+				case storageV1.ReadWritePermission:
+					policy.Statement = append(policy.Statement, read, write)
+				default:
+					return "", fmt.Errorf("unknown permission, %s", *spec.LocalPermission)
+				}
+			}
+		*/
+		policy.Statement = append(policy.Statement, read, write)
+	}
 
-	b, err := json.Marshal(&policy)
+	b, err := json.MarshalIndent(&policy, "", "  ")
```

Comment:
Does this change have any material effect?

Reply:
No, but it makes it easier to read the policy from the CMC